wweval/R/eval_postprocess.R

```diff
@@ -0,0 +1,548 @@
+    forecast_date = forecast_date,
+    model_type = model,
+    location = location
+  )
+
+  save_table(
+    data_to_save = full_ww_quantiles,
+    type_of_output = "ww_quantiles",
+    output_dir = output_dir,
+    scenario = scenario,
+    forecast_date = forecast_date,
+    model_type = model,
+    location = location
+  )
+
+  ### Plot the draw comparison-------------------------------------
+  plot_hosp_draws <- {
+    if (is.null(hosp_draws)) {
+      NULL
+    } else {
+      get_plot_hosp_data_comparison(
+        hosp_draws,
+        location,
+        model_type = model
+      )
+    }
+  }
+
+  hosp_draw_plot_savename <- ifelse(
+    ww_model,
+    "plot_hosp_draws",
+    "plot_hosp_draws_hosp_model"
+  )
+
+  ggsave_plot(plot_hosp_draws,
+    save_basename = hosp_draw_plot_savename
+  )
+  save_object(plot_hosp_draws)
+
+  plot_hosp_t <- make_fig2_hosp_t(
+    hosp_quantiles = full_hosp_quantiles,
+    loc_to_plot = location,
+    date_to_plot = forecast_date
+  ) +
+    ggtitle(glue::glue("{location} on {forecast_date}")) +
+    theme_bw()
+
+  ggsave_plot(plot_hosp_t)
+
+
+  # Plots of R(t)
+  draws <- wwinference::get_draws(wwinference_fit_obj, what = "all")
+
+  plot_state_rt <- wwinference::get_plot_global_rt(
+    draws$global_rt,
+    forecast_date
+  )
+  ggsave_plot(plot_state_rt)
+
+  if (ww_model) {
+    plot_subpop_rt <- wwinference::get_plot_subpop_rt(
+      draws$subpop_rt,
+      forecast_date
+    )
+    ggsave_plot(plot_subpop_rt)
+  }
+
+  if (!is.null(ww_draws)) {
+    plot_ww_draws <- get_plot_ww_data_comparison(
+      ww_draws,
+      location,
+      model_type = model
+    )
+
+    ggsave_plot(plot_ww_draws)
+  } else {
+    plot_ww_draws <- NULL
+  }
+  save_object(plot_ww_draws)
+
+  if (!is.null(full_ww_quantiles)) {
+    plot_ww_t <- make_fig2_ct(
+      full_ww_quantiles,
+      loc_to_plot = location,
+      date_to_plot = forecast_date,
+      max_n_site_labs_to_show = length(unique(full_ww_quantiles$lab_site_index))
+    ) +
+      facet_wrap(~site_lab_name, scales = "free_y") +
+      ggtitle(glue::glue("{location} on {forecast_date}")) +
+      theme_bw()
+
+    ggsave_plot(plot_ww_t)
+  }
+
+
+  ## Score hospital admissions forecasts----------------------------------
+  hosp_scores <- get_full_scores(hosp_draws, scenario)
+  save_object(hosp_scores)
+  save_table(
+    data_to_save = hosp_scores,
+    type_of_output = "scores",
+    output_dir = output_dir,
+    scenario = scenario,
+    forecast_date = forecast_date,
+    model_type = model,
+    location = location
+  )
+  hosp_scores_quantiles <- get_scores_from_quantiles(hosp_quantiles, scenario)
+  save_object(hosp_scores_quantiles)
+  save_table(
+    data_to_save = hosp_scores_quantiles,
+    type_of_output = "scores_quantiles",
+    output_dir = output_dir,
+    scenario = scenario,
+    forecast_date = forecast_date,
+    model_type = model,
+    location = location
+  )
+}
+
+#' Postprocess model for evaluation.
+#'
```

Comment:
Would explain where things get saved in this (in nested file structure) and what gets saved, as well s the dependency on `postprocess_successful_fit()`